cmake/modules/CTags.cmake

```diff
@@ -0,0 +1,26 @@
+# after https://www.topbug.net/blog/2012/03/17/generate-ctags-files-for-c-slash-c-plus-plus-source-files-and-all-of-their-included-header-files/ and
+# https://stackoverflow.com/questions/9827208/run-a-shell-command-ctags-in-cmake-and-make
+
+set_source_files_properties(tags PROPERTIES GENERATED true)
+add_custom_target(ctags
```

Comment:
if we want to make this a cmake module, maybe we should make it more generalized. i.e. something like: 
```cmake
execute_process(
  COMMAND git config --file .gitmodules --get-regexp path
  COMMAND awk '/src/ { print $2 }'
  OUTPUT_VARIABLE submodules
  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
set(excludes "--exclude=*.js")
foreach(submodule ${submodules}) 
  set(excludes "${excludes} --exclude=${submodule}/*")
endforeach()
add_custom_target(ctags
  VERBATIM
  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src
  COMMENT "Build ctags file src/tags (no submodules)"
  ctags -R --c++-kinds=+p --fields=+iaS --extra=+q ${excludes})
```

/// disclaimer: not tested.